### DISCUSSION TITLE
Reset does not work as expected

### DISCUSSION THREAD

**Discussion by mansoorkhan96** [livewire/livewire#6753] [7. Bugs (archived)] [answered]
### Livewire version

v3

### Which PHP version are you using?

PHP 8.1

### Steps To Reproduce

1. Set an int property on your livewire component.

```php
#[Url]
public ?int $category_id = null;
```

2. Add an action method and a button to reset the property

```php
<button wire:click.prevent="clearAll" >Clear All</button>
```

```php
public function clearAll()
{
    $this->reset('category_id');
}
```

3. You see an exception: 
`Livewire\Exceptions\PropertyNotFoundException`
`Property [$category_id] not found on component: [activities.category]`

### Additional information

Resetting a Livewire int property seems to be not working as expected. calling $this->reset('propertyName') throws `Livewire \Exception \PropertyNotFoundException` exception.

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by PhiloNL** [comment]
I'm unable to replicate this, the following component works just fine:

```php
class Users extends Component
{
    #[Url]
    public ?int $number = null;

    public function clear()
    {
        $this->reset('number');
    }

    public function render()
    {
        return <<<'BLADE'
            <div>
                <div>Number: {{ $number }}</div>
                <input type="text"  wire:model.live="number">
                <button wire:click="clear">Clear</button>
            </div>
        BLADE;
    }
}
```

Are you using the latest v3 release?

**Reply by whousehomestay** [reply]
Maybe this: https://github.com/livewire/livewire/commit/782ffbfb33b533d53e0e44daeb197b94e688ebba ?

**Reply by PhiloNL** [reply, marked as answer]
Yeah, that might be it, I'm on the master branch haha. So at least we know it's fixed in the next release 😄

**Reply by mansoorkhan96** [reply]
Hey @PhiloNL 
It seems to be fixed in new release i.e. `v3.0.2` Thank you very much.

Sorry for taking days. Github never notified me for replies.

**Comment by yrizzz** [comment]
<img width="779" height="291" alt="lv" src="https://github.com/user-attachments/assets/e16c46cd-68e6-4b66-ad48-52f52c7a8d85" />

find this file 
vendor\livewire\livewire\src\Concerns\InteractsWithProperties.php check line start 60, modify that method same in the picture
